### PR TITLE
David physics tmp

### PR DESCRIPTION
**Pull request by drichte2**
The next installment of the large PR. This one adds a new type of droplet statistic (droplet histograms) into the droplet_diag output.

### CODE REVIEW

File: src/droplet.F [hunk 1]
```diff
@@ -1645,6 +1644,12 @@ subroutine droplet_diag(rtime,nrec,zh,zf,pdata,pdata_locind,dpten)
       integer :: tnumpart,tnumdrop,tnumaerosol
       real :: radavg,radsqr,radmin,radmax,Tpavg,Tpmin,Tpmax,Tpsqr,Tfavg,qfavg,prsavg,rhoavg
       real :: partnum(nk),numconc(nk),Tpsrc(nk),qvsrc(nk),qf(nk),Tf(nk),radmean(nk),Tpmean(nk),qfsat(nk)
+
+      integer, parameter :: nbins = 514
```
Comment (johnmauff) [inline src/droplet.F]: I would like to see what the mean of nbins described in a comment.

**Review by johnmauff — APPROVED** [review]
@drichte2 This is exactly the size of PR that make review simple and fast.  Thank you for making it a manageable size.  Other than the need to add a comment about the meaning of nbins=514 I think this PR is ready.

File: src/droplet.F [hunk 2]
```diff
@@ -1932,15 +1941,83 @@ subroutine droplet_diag(rtime,nrec,zh,zf,pdata,pdata_locind,dpten)
...
+
+      dr = (rmax10-rmin10)/(nbins-1)
+
+      bins_radius(1) = rmin10-dr
+      do i=1,nbins-1
+         bins_radius(i+1) = bins_radius(i) + dr
+      end do 
+
+      !Now that the bin centers are defined, loop through particles
+      do np=1,nparcelsLocalActive
```
Comment (sjsprecious) [inline src/droplet.F:1966]: either need an OpenACC directive for this loop or update`pdata` on the host first.

File: src/droplet.F [hunk 3]
```diff
@@ -1932,15 +1941,83 @@ subroutine droplet_diag(rtime,nrec,zh,zf,pdata,pdata_locind,dpten)
...
+#endif
+
 #ifdef NETCDF
       ! JS: this subroutine is not updated with the new MPI
       !     interface for the droplets yet
 #ifdef MPI
       if ( myid == 0 ) then
 #endif
-      call writedropdiag_nc(nrec, &
+      call writedropdiag_nc(nrec,nbins, &
```
Comment (sjsprecious) [inline src/droplet.F:2015]: so the `hist_radius10` and `hist_radius` are always written to the netCDF file? In this case, we could not check its values & correctness through the log files.

**Review by sjsprecious — COMMENTED** [review]
@drichte2 , thanks for this PR and it generally looks good to me, too. I have a quick question about how to evaluate / check the new droplet diagnostics.

File: src/droplet.F (hunk 3 above)
Comment (drichte2) [reply src/droplet.F:2015]: That's right, however the average droplet radius is already being compared, which is the mean of this distribution. In other words, if the average droplet radius is checking out, then that must mean that the histogram is also okay. We could always add a write statement within this routine if needed, but the bigger thing is to figure out where to put the droplet_diag in the future

Comment (sjsprecious) [reply src/droplet.F:2015]: Thanks for your clarification. That is clear to me now.

**Review by sjsprecious — APPROVED** [review]

**Pull request merged**